src/network/core/packet.cpp

```diff
@@ -38,16 +38,24 @@ Packet::Packet(NetworkSocketHandler *cs, size_t limit, size_t initial_read_size)
 
 /**
  * Creates a packet to send
+ * @param cs    The socket handler associated with the socket we are writing to; could be \c nullptr.
  * @param type  The type of the packet to send
  * @param limit The maximum number of bytes the packet may have. Default is COMPAT_MTU.
  *              Be careful of compatibility with older clients/servers when changing
  *              the limit as it might break things if the other side is not expecting
  *              much larger packets than what they support.
  */
-Packet::Packet(PacketType type, size_t limit) : pos(0), limit(limit), cs(nullptr)
+Packet::Packet(NetworkSocketHandler *cs, PacketType type, size_t limit) : pos(0), limit(limit), cs(cs)
 {
 	/* Allocate space for the the size so we can write that in just before sending the packet. */
-	this->Send_uint16(0);
+	size_t size = sizeof(PacketSize);
+	if (cs != nullptr && cs->send_encryption_handler != nullptr) {
+		/* Allocate some space for the message authentication code of the encryption. */
+		size += cs->send_encryption_handler->MACSize();
```

Comment:
Shouldn't MTU also be adjusted for that, like in https://github.com/Milek7/OpenTTD/commit/d3653710d35e44c63397601e67c31d2c10f9a0ae#diff-31dd9b11e3c119e4e65cf2b3dc0ab704b71d41728d0e1ad8ecc109cfee907ed7L177?

Reply:
But it does increase real packet size on the wire. Though that doesn't really matter for TCP and indeed I noticed now that in the meantime MTU for TCP was split and increased anyway, so it's fine.

Reply:
I'm not understanding your concern with the MTU.

The packet itself limits the size to some limit, be it the old 1460 MTU or 32k. So, the maximum size for packets that do regularly reach it, i.e. the map packets, has not changed at all.
Due to the MAC there will be more data, so the last packet might be bigger and/or there might be more packets but that has nothing to do with having to adjust the MTU.

In the context of your example, I have already increased `this->current->size` by `MACSize` by the point you get to that selected line of code. So the MTU does not need to be amended.

Reply:
I don't have concern with MTU now. (it was under assumption that low 1460 value was there for a specific reason, but that's not relevant anymore)